src/Sulu/Bundle/SecurityBundle/Controller/UserController.php

```diff
@@ -313,7 +313,18 @@ public function cgetAction(Request $request)
                 $user = $this->getDoctrine()->getRepository(
                     $this->container->getParameter('sulu.model.user.class')
                 )->findUserByContact($contactId);
-                $view = $this->view($user ?? new \stdClass(), 200);
+
+                if (!$user) {
+                    $contact = $this->getDoctrine()->getRepository(
+                        $this->container->getParameter('sulu.model.contact.class')
+                    )->findById($contactId);
+
+                    $user = [
+                        'contact' => $contact,
```

Comment:
That part is also a bit hard to solve... The thing is that we need to add a contact id to the response, in the following format, so that a part of the request has the following format:

```javascript
{
    contact: {
        id: 1
    },
    username: 'bla'
}
```

Since that is not easily possible in our generic form I have decided to also return that part in this response even if there is no user. Therefore I have to create an array faking this object, which makes it so weird.

In the [old UI it was simply reusing the already loaded contact for that](https://github.com/sulu/sulu/blob/abf10c8/src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/users/main.js#L55). That's a viable solution, which would probably need us to make a custom view.

Reply:
I've had an idea last night about this... The real problem is that we can't really predict what the static key `contact` should be in the JSON we must send, at least not without introducing another parameter which will make the component even harder to use. What if we make a slight change to the API instead? The problem is especially on the `POST` request, and we could probably also model it something like this:

```
POST /admin/api/users?contactId=5
```

Then we could say if the `Form` react component gets an `idQueryParameter` passed we simply add this to the request, and the backend can work with that if required.

@chirimoya WDYT?